src/routes/index.js

```diff
@@ -239,14 +241,123 @@ router.get('/security', (req, res) => {
   res.render('security', { currentUser: req.user });
 });
 
-router.get('/troubleshooting', (req, res) => {
-  res.render('troubleshooting', { currentUser: req.user });
-});
-
 router.get('/statistics', (req, res) => {
   res.render('statistics', { currentUser: req.user, headerActive: 'stats' });
 });
 
+router.get('/resetPassword', (req, res) => {
+  res.render('resetPassword', { platform: process.env.ENV });
+});
+
+router.post('/resetPassword', registerLimiter, async (req, res) => {
+  // if we are local, we can skip the captcha
+  if (process.env.ENV === 'prod') {
+    req.body.captcha = req.body['g-recaptcha-response'];
+    if (
+      req.body.captcha === undefined ||
+      req.body.captcha === '' ||
+      req.body.captcha === null
+    ) {
+      req.flash('error', 'The captcha failed or was not inputted.');
+      res.redirect('/resetPassword');
+      return;
+    }
+
+    const secretKey = process.env.MY_SECRET_GOOGLE_CAPTCHA_KEY;
+
+    const verifyUrl = `https://google.com/recaptcha/api/siteverify?secret=${secretKey}&response=${req.body.captcha}&remoteip=${req.connection.remoteAddress}`;
+    const body = await request(verifyUrl);
+
+    if (body.success !== undefined && !body.success) {
+      req.flash('error', 'Failed captcha verification.');
+      res.redirect('/resetPassword');
+      return;
+    }
+  }
+
+  const secretKey = process.env.MY_SECRET_GOOGLE_CAPTCHA_KEY;
+
+  const verifyUrl = `https://google.com/recaptcha/api/siteverify?secret=${secretKey}&response=${req.body.captcha}&remoteip=${req.connection.remoteAddress}`;
```

Comment:
Would be nice to extract this out to reduce duplication but let's leave it for now so you can get your change out sooner.